src/client.cpp

```diff
@@ -1257,15 +1257,15 @@ void CClient::ProcessAudioDataIntern ( CVector<int16_t>& vecsStereoSndCrd )
             if ( bMuteOutStream )
             {
                 iUnused = opus_custom_encode ( CurOpusEncoder,
-                                               &vecZeros[i * iNumAudioChannels * iOPUSFrameSizeSamples],
+                                               &vecZeros[static_cast<size_t> ( i * iNumAudioChannels * iOPUSFrameSizeSamples )],
```

Comment:
We should calculate an upper bound of these and check if that's valid.

Reply:
`i < iSndCrdFrameSizeFactor` which is a small number (usually 1, 2 or 4).
`iNumAudioChannels` is 1 or 2.
`SYSTEM_FRAME_SIZE_SAMPLES` is 64 and `iOPUSFrameSizeSamples` is either that or twice that.

So the result should be < 16K.

Reply:
Exactly, which illustrates why the CodeQL warning is silly in a lot of contexts, including this one.

I have mentioned an alternative way to fix it in the comments on #3161, and have what I think is an even better way just compiling and about to push to a branch.

Reply:
We're more likely to see buffer overruns -- but we don't.

`vecZeros` and `vecsStereoSndCrd` are 2 * `Sound.Init ( iPrefMonoFrameSize )`, so likely to be < 16K as well.

Reply:
The CodeQL warnings we are addressing are nothing to do with buffer overruns. Just a perceived arithmetic overflow that will never happen with the values we are using.

Reply:
With `i`, `iNumAudioChannels` and `iOPUSFrameSizeSamples` as `int`, however, MAX_INT * MAX_INT * MAX_INT would be out of bounds for the index type.